### ISSUE TITLE
Installation of netcdf4 fails

### ISSUE THREAD

**Issue by heistermann** [openradar/oss_weather_radar_vm#33]
During provisioning in `install_scipy_stack.sh`, the command `sudo pip install netcdf4` fails. See report below. Although a lot of hdf5 related "not found" message occur, the actual error is shown at the end:

```
ImportError: <module 'setuptools.dist' from '/usr/lib/python2.7/dist-packages/se
tuptools/dist.pyc'> has no 'check_specifier' attribute
```

Here's the log:
```
Downloading/unpacking netcdf4
  Downloading netCDF4-1.3.1.tar.gz (548kB): 548kB downloaded
  Running setup.py (path:/tmp/pip_build_root/netcdf4/setup.py) egg_info for pack
age netcdf4
    Package hdf5 was not found in the pkg-config search path.
    Perhaps you should add the directory containing `hdf5.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'hdf5' found
    /usr/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution
option: 'long_description_content_type'
      warnings.warn(msg)
    using nc-config ...

        HDF5_DIR environment variable not set, checking some standard locations
..
    checking /home/vagrant ...
    checking /usr/local ...
    checking /sw ...
    checking /opt ...
    checking /opt/local ...
    checking /usr ...
    HDF5 library version: 1.8.11 found in /usr
    using netcdf library version 4.1.3
    using Cython to compile netCDF4.pyx...
    netcdf lib does not have group rename capability
    netcdf lib has nc_inq_path function
    netcdf lib does not have nc_inq_format_extended function
    netcdf lib does not have nc_open_mem function
    netcdf lib does not have cdf-5 format capability
    netcdf lib does not have netcdf4 parallel functions
    Traceback (most recent call last):
      File "<string>", line 17, in <module>
      File "/tmp/pip_build_root/netcdf4/setup.py", line 583, in <module>
        **setuptools_extra_kwargs)
      File "/usr/lib/python2.7/distutils/core.py", line 111, in setup
        _setup_distribution = dist = klass(attrs)
      File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 239, in _
_init__
        self.fetch_build_eggs(attrs.pop('setup_requires'))
      File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 264, in f
etch_build_eggs
        replace_conflicting=True
      File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 620, in res
olve
        dist = best[req.key] = env.best_match(req, ws, installer)
      File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 858, in bes
t_match
        return self.obtain(req, installer) # try and download/install
      File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 870, in obt
ain
        return installer(requirement)
      File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 314, in f
etch_build_egg
        return cmd.easy_install(req)
      File "/usr/lib/python2.7/dist-packages/setuptools/command/easy_install.py"
, line 616, in easy_install
        return self.install_item(spec, dist.location, tmpdir, deps)
      File "/usr/lib/python2.7/dist-packages/setuptools/command/easy_install.py"
, line 646, in install_item
        dists = self.install_eggs(spec, download, tmpdir)
      File "/usr/lib/python2.7/dist-packages/setuptools/command/easy_install.py"
, line 834, in install_eggs
        return self.build_and_install(setup_script, setup_base)
      File "/usr/lib/python2.7/dist-packages/setuptools/command/easy_install.py"
, line 1040, in build_and_install
        self.run_setup(setup_script, setup_base, args)
      File "/usr/lib/python2.7/dist-packages/setuptools/command/easy_install.py"
, line 1025, in run_setup
        run_setup(setup_script, args)
      File "/usr/lib/python2.7/dist-packages/setuptools/sandbox.py", line 50, in
 run_setup
        lambda: execfile(
      File "/usr/lib/python2.7/dist-packages/setuptools/sandbox.py", line 100, i
n run
        return func()
      File "/usr/lib/python2.7/dist-packages/setuptools/sandbox.py", line 52, in
 <lambda>
        {'__file__':setup_script, '__name__':'__main__'}
      File "setup.py", line 187, in <module>
        except:
      File "/usr/lib/python2.7/distutils/core.py", line 111, in setup
        _setup_distribution = dist = klass(attrs)
      File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 243, in _
_init__
        _Distribution.__init__(self,attrs)
      File "/usr/lib/python2.7/distutils/dist.py", line 287, in __init__
        self.finalize_options()
      File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 277, in f
inalize_options
        ep.load()(self, ep.name, value)
      File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2093, in lo
ad
        raise ImportError("%r has no %r attribute" % (entry,attr))
    ImportError: <module 'setuptools.dist' from '/usr/lib/python2.7/dist-package
s/setuptools/dist.pyc'> has no 'check_specifier' attribute
    Complete output from command python setup.py egg_info:
    Package hdf5 was not found in the pkg-config search path.

Perhaps you should add the directory containing `hdf5.pc'

to the PKG_CONFIG_PATH environment variable

No package 'hdf5' found

/usr/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution opti
on: 'long_description_content_type'

  warnings.warn(msg)

reading from setup.cfg...

using nc-config ...



    HDF5_DIR environment variable not set, checking some standard locations ..

checking /home/vagrant ...

checking /usr/local ...

checking /sw ...

checking /opt ...

checking /opt/local ...

checking /usr ...

HDF5 library version: 1.8.11 found in /usr

using netcdf library version 4.1.3

using Cython to compile netCDF4.pyx...

netcdf lib does not have group rename capability

netcdf lib has nc_inq_path function

netcdf lib does not have nc_inq_format_extended function

netcdf lib does not have nc_open_mem function

netcdf lib does not have cdf-5 format capability

netcdf lib does not have netcdf4 parallel functions

Traceback (most recent call last):

  File "<string>", line 17, in <module>

  File "/tmp/pip_build_root/netcdf4/setup.py", line 583, in <module>

    **setuptools_extra_kwargs)

  File "/usr/lib/python2.7/distutils/core.py", line 111, in setup

    _setup_distribution = dist = klass(attrs)

  File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 239, in __ini
t__

    self.fetch_build_eggs(attrs.pop('setup_requires'))

  File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 264, in fetch
_build_eggs

    replace_conflicting=True

  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 620, in resolve


    dist = best[req.key] = env.best_match(req, ws, installer)

  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 858, in best_ma
tch

    return self.obtain(req, installer) # try and download/install

  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 870, in obtain

    return installer(requirement)

  File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 314, in fetch
_build_egg

    return cmd.easy_install(req)

  File "/usr/lib/python2.7/dist-packages/setuptools/command/easy_install.py", li
ne 616, in easy_install

    return self.install_item(spec, dist.location, tmpdir, deps)

  File "/usr/lib/python2.7/dist-packages/setuptools/command/easy_install.py", li
ne 646, in install_item

    dists = self.install_eggs(spec, download, tmpdir)

  File "/usr/lib/python2.7/dist-packages/setuptools/command/easy_install.py", li
ne 834, in install_eggs

    return self.build_and_install(setup_script, setup_base)

  File "/usr/lib/python2.7/dist-packages/setuptools/command/easy_install.py", li
ne 1040, in build_and_install

    self.run_setup(setup_script, setup_base, args)

  File "/usr/lib/python2.7/dist-packages/setuptools/command/easy_install.py", li
ne 1025, in run_setup

    run_setup(setup_script, args)

  File "/usr/lib/python2.7/dist-packages/setuptools/sandbox.py", line 50, in run
_setup

    lambda: execfile(

  File "/usr/lib/python2.7/dist-packages/setuptools/sandbox.py", line 100, in ru
n

    return func()

  File "/usr/lib/python2.7/dist-packages/setuptools/sandbox.py", line 52, in <la
mbda>

    {'__file__':setup_script, '__name__':'__main__'}

  File "setup.py", line 187, in <module>

    except:

  File "/usr/lib/python2.7/distutils/core.py", line 111, in setup

    _setup_distribution = dist = klass(attrs)

  File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 243, in __ini
t__

    _Distribution.__init__(self,attrs)

  File "/usr/lib/python2.7/distutils/dist.py", line 287, in __init__

    self.finalize_options()

  File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 277, in final
ize_options

    ep.load()(self, ep.name, value)

  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2093, in load

    raise ImportError("%r has no %r attribute" % (entry,attr))

ImportError: <module 'setuptools.dist' from '/usr/lib/python2.7/dist-packages/se
tuptools/dist.pyc'> has no 'check_specifier' attribute
```

**Comment by heistermann** [comment]
Having `sudo pip install -U pip` before `sudo pip install netcdf4` appears to resolve the issue. PR follows.

**Comment by heistermann** [comment]
Fixed by PR #34.
